File: src/edu/gbcg/dbInteraction/dbSelector/reddit/submissions/SubmissionSetMapper.java
package edu.gbcg.dbInteraction.dbSelector.reddit.submissions;

import edu.gbcg.dbInteraction.dbSelector.RSMapper;
import edu.gbcg.dbInteraction.dbcreator.reddit.submissions.SubmissionsFacilitator;

import java.sql.ResultSet;
import java.util.List;
import java.util.Map;

/**
 * Specific RSMapper object for a specific DB type. The implementation for buildMappers is handled in the parent
 * abstract class, however it needs to know the column names for this specific DB so pass it the column names and let
 * it work
 */
public class SubmissionSetMapper extends RSMapper {
    public SubmissionSetMapper(Map<String, String> map){ super(map); }
    public SubmissionSetMapper(){ super(); }
    public List<RSMapper> buildMappers(ResultSet rs){
        return buildMappers_impl(rs, SubmissionsFacilitator.getColumnsForDB());
    }
}

File: src/main/kotlin/edu/gbcg/dbInteraction/dbSelector/reddit/submissions/SubmissionSetMapper.kt
/*
 * Copyright (c) 2018 Sean Grimes. All Rights Reserved.
 * License: MIT
 */

package edu.gbcg.dbInteraction.dbSelector.reddit.submissions

import edu.gbcg.configs.columnsAndKeys.RedditSubs
import edu.gbcg.dbInteraction.dbSelector.RSMapper
import org.json.JSONObject
import java.sql.ResultSet

class SubmissionSetMapper: RSMapper {
    constructor(map: Map<String, String>): super(map)
    constructor(jsonObject: JSONObject): super(jsonObject)
    constructor(): super()
    override fun buildMappers(rs: ResultSet): MutableList<RSMapper>? {
        return buildMappersImpl(rs, RedditSubs.columnNames())
    }
}

